libcnb-data/src/buildpack_plan.rs

```diff
@@ -50,4 +64,24 @@ name = "rust"
         let result = toml::from_str::<BuildpackPlan>(toml);
         assert!(result.is_ok());
     }
+
+    #[test]
+    fn it_deserializes_metadata() {
+        #[derive(Deserialize)]
+        struct Metadata {
+            foo: String,
+        }
+
+        let mut metadata = Table::new();
+        metadata.insert("foo".to_string(), toml::Value::String("bar".to_string()));
+        let entry = Entry {
+            name: "foo".to_string(),
+            metadata,
+        };
+
+        let result = entry.metadata::<Metadata>();
+        assert!(result.is_ok());
+        let metadata = result.unwrap();
+        assert_eq!(metadata.foo, "bar".to_string());
```

Comment:
Nit (doesn't need to be changed): This could be written like so if we change the attribute of `Metadata` to `#[derive(Deserialize, Eq, PartialEq, Debug)]`:

```rust
 assert_eq!(
    entry.metadata(),
    Ok(Metadata {
        foo: String::from("bar")
    })
);
```

Test failures would be little easier to read since we would see the whole expected and actual structure in the test output.

Reply:
@Malax i feel like a lot of things we write should probably have these things derived on them in `libcnb-data`